### DISCUSSION TITLE
ARM function

### DISCUSSION THREAD

**Discussion by m-rik-prog** [ExpressLRS/ExpressLRS#3005] [General]
Good morning,

I need some clarification from you on the ARM function when in ‘16ch Rate/2’ mode at 333Mhz whereby channel 5 can take on all intermediate values between 732us and 2268us. Clarification that I seem to be missing (or am I missing it) in the documentation.

You write that when channel 5 is at 1000us the ARM function is off and when it is at 2000us the ARM function is on.
However, you do not say (at least I have not found it) what happens with all intermediate values between 1001us and 1999us and with the lower values between 732us and 999us and with the higher values between 2001us and 2268us.

In drones, if the gyroscopes see any rotation of the model, they can start the motors and for this reason it is essential to be able to disarm the motors so that moving the drone does not cause the motors to start causing damage.

On the other hand, in RC helicopters the ARM function has never been necessary because the gyroscopes do not control the motors as they do in drones but only control the servos that move the swashplate and even when the ESC is armed but Throttle Cut and/or Throttle Hold are active any movement of the heli can never start the motor.

I would therefore like to understand if unlike all the other protocols (including TBS) I have to write in the instructions for ELRS users that channel 5 can NEVER be used even with extended range (16ch Rate/2) but that it must be set to the fixed and precise value of 2000us otherwise, as you explain in the documentation, the module cooling fan may not start, burning the module and the module power control may not work with the risk of losing control of the model (due to signal saturation or loss).
I also understand that in the Init of the ELRS versions of the LUA applications for integrating my control units with the transmitters I have to set channel 5 to 1000us for maximum performance.

In my control units to exploit all the current possible functions, up to 13 channels can be used today, but I already plan to use three more channels for new functions in the future. Not being able to use one channel (channel 5) freely is a pity.
Don't you exist and/or don't you plan to create a configuration (or an option) that disables the ARM function so that you can use ELRS like all other protocols?

Thank you

**Comment by mha1** [comment]
ch5 > 1500us -> armed, disarmed otherwise

**Reply by m-rik-prog** [reply]
Thank you!
I suggest adding a reference to a note for the 16ch Rate/2 configuration in the table of various configurations.
For other configurations where channel 5 can only take on two values it is not needed

**Comment by mha1** [comment]
> Don't you exist and/or don't you plan to create a configuration (or an option) that disables the ARM function so that you can use ELRS like all other protocols?

work in progress, nothing decided yet

**Reply by m-rik-prog** [reply]
Thank you!
It would make sense and would only be VERY useful in the 16ch Rate/2 configuration.
In all other configurations where channel 5 has only two possible values, it would be of no use.

**Reply by mha1** [reply]
> Thank you! It would make sense and would only be useful in the 16ch Rate/2 configuration. In all other configurations where channel 5 has only two possible values, it would be of no use.

8ch and 12ch/mixed can be made to also output ch5 as proportional channel, just like 16ch/half rate.

**Reply by inventor7777** [reply]
This might not be the newest arming discussion, but @m-rik-prog echoes the exact thing I’ve been thinking. 

As a surface user, there’s not much more frustrating to have channel 5 taken up completely by a function that’s not necessary on surface. It would be nice to simply be able to move the arm function, or **simply have a simple option to disable it altogether and get channel 5 back.** (or even assign arming to a software channel like 0)

I guess for now I can remap so that channel 5 pins are just remapped to another channel, but it makes it unnecessarily complex and frustrating to look at the channel monitor trying to remember what goes to what. 

I suppose there’s different mapping modes like Hybrid, but that makes it even more confusing.

@mha1

**Reply by mha1** [reply]
V4 will provide a solution, see https://github.com/ExpressLRS/ExpressLRS/pull/3008

**Reply by inventor7777** [reply]
Very nice, thank you for info @mha1.

Do you know any information about when 4.0 will be released and available for download? Even as a beta.

**Reply by mha1** [reply]
There's no release date for V4 yet but you can test V4 by flashing your receiver and module to the ELRS master branch (V4 development). To do this use the Configurator and enable Expert Mode in the Configurator Settings. Then leave the Settings and select Configurator. Select GIT BRANCH and select master. Populate the rest as you normally do. EdgeTX 2.11 has already released the portion that complements the ELRS V4 arming side so just put the latest release of EdgeTX 2.11 on your radio.

**Reply by inventor7777** [reply]
Awesome, thanks! I will try that once I test the current setup (I’m brand new to ELRS) and if it’s good I will likely try V4.

**Comment by m-rik-prog** [comment]
![image](https://github.com/user-attachments/assets/b66d62a2-b746-4438-ab95-073380d3dd99)

**Reply by mha1** [reply]
I know. What I'm saying is this can be changed

**Reply by m-rik-prog** [reply]
Fine, but 8 channels and 12 channels (minus channel 5 used for ARM function) would still be few compared to the 13 that may all be needed.
Even Spektrum released AirWare versions in August that double the number of channels of their transmitters. Also Futaba announced new 24-channel transmitters

**Comment by mha1** [comment]
all modes have up- and downsides
- 8ch: only 8 channels but at full 100Hz or 333Hz
- 12ch/mixed: (only) 12 channels but first four at full rate, 5-12 at half rate
- 16ch/half rate: 16 channels (yay) but all on half rate.

Why are you talking about 13 channels? 16ch/half rate can do 16 channels.

**Reply by m-rik-prog** [reply]
Thank you, but I had only mentioned the 16ch Rate/2 protocol from the start. Of the 8 and 12 channels I had not mentioned it

**Comment by mha1** [comment]
"In all other configurations where channel 5 has only two possible values, it would be of no use."

I just referred to your comment. Having a proportional channel 5 in 8ch and 12ch modes too would be of use for many other LOS setups where you don't need so many channels but are able to benefit from the full packet rate update rate.
